prowler/providers/azure/services/network/network_watcher_enabled/network_watcher_enabled.py

```diff
@@ -5,24 +5,25 @@
 class network_watcher_enabled(Check):
     def execute(self) -> Check_Report_Azure:
         findings = []
-        nw_locations = []
         for subscription, network_watchers in network_client.network_watchers.items():
-            for network_watcher in network_watchers:
-                nw_locations.append(network_watcher.location)
-        for subscription, locations in network_client.locations.items():
-            for location in locations:
-                report = Check_Report_Azure(self.metadata())
-                report.subscription = subscription
-                report.resource_name = "Network Watcher"
-                report.location = location
-                report.resource_id = f"/subscriptions/{subscription}/providers/Microsoft.Network/networkWatchers/{location}"
-                if location not in nw_locations:
-                    report.status = "FAIL"
-                    report.status_extended = f"Network Watcher is not enabled for the location {location} in subscription {subscription}."
-                    findings.append(report)
-                else:
-                    report.status = "PASS"
-                    report.status_extended = f"Network Watcher is enabled for the location {location} in subscription {subscription}."
-                    findings.append(report)
+            report = Check_Report_Azure(self.metadata())
+            report.subscription = subscription
+            report.resource_name = "Network Watcher"
+            report.location = "Global"
+            report.resource_id = f"/subscriptions/{network_client.subscriptions[subscription]}/resourceGroups/NetworkWatcherRG/providers/Microsoft.Network/networkWatchers/NetworkWatcher_*"
+            report.status = "FAIL"
+            report.status_extended = f"Network Watcher is not enabled for all locations in subscription '{subscription}'."
```

Comment:
Would it be possible to indicate which are the locations that do not have network watchers?

Reply:
Good point, thanks to that I start thinking in how to do and I believe that I improve so much the business logic of the check in terms of efficiency and readability. Review it again and say me if you see better than before.